var/spack/repos/builtin/packages/cusz/package.py

```diff
@@ -6,19 +6,22 @@
 from spack.package import *
 
 
-class Cusz(MakefilePackage):
-    """cuSZ is a CUDA-based error-bounded lossy compressor for scientific
-    data (floating point and integers).
-    """
-
-    homepage = "https://szcompressor.org"
-    url = "https://github.com/szcompressor/cuSZ/releases/download/v0.1.2/cuSZ-0.1.2.tar.gz"
-    git = "https://github.com/szcompressor/cuSZ"
-    maintainers = ["dingwentao", "jtian0"]
-
-    version("master", branch="master")
-    version("0.1.2", sha256="c6e89a26b295724edefc8052f62653c5a315c78eaf6d5273299a8e11a5cf7363")
-
-    def install(self, spec, prefix):
-        mkdir(prefix.bin)
-        install("bin/cusz", prefix.bin)
+class Cusz(CMakePackage, CudaPackage):
+    """A GPU accelerated error-bounded lossy compression for scientific data"""
+
+    homepage = "https://szcompressor.org/"
+    git = "https://github.com/robertu94/cusz"
+
+    maintainers = ["jtian0", "dingwentao"]
+
+    conflicts("~cuda")
+    conflicts("cuda_arch=none", when="+cuda")
+
+    version("develop", branch="develop")
```

Comment:
In the past, I've had issues getting the older versions to build consistently.  I suspect that people aren't using it widely yet.

>  any recent fixed version ... would be ideal since `@develop` isn't reproducible.

@jtian0 or @dingwentao  Could you please make a release of cuSZ containing the fixes I sent this spring so I can add it to this PR.

Reply:
done for now.  We'll send the HIP stuff in a later PR